testSettings.py

```diff
@@ -37,9 +35,42 @@ def _post_check(self) -> None:
         self._lock: threading.Lock
         object.__setattr__(self, "_lock", threading.Lock())
 
-        # Check that the cfg_descr has a connection/test_case_id
+        # Access some properties here. We would get an exception early on, if
+        # there is something wrong.
         self.connection
-        self.test_case_id
+        self.test_case_id.info
+        self.conf_server
+        self.conf_client
+
+    @property
+    def conf_server(self) -> testConfig.ConfServer:
+        # For now, only one server/client is supported and TestConfig already
+        # enforces that. Do tuple-unpacking here, to further assert that there
+        # is only one server/client.
+        (c_server,) = self.connection.server
+        return c_server
+
+    @property
+    def conf_client(self) -> testConfig.ConfClient:
+        # For now, only one server/client is supported and TestConfig already
+        # enforces that. Do tuple-unpacking here, to further assert that there
+        # is only one server/client.
+        (c_client,) = self.connection.client
+        return c_client
+
+    @property
+    def conf_server_used(self) -> testConfig.ConfBaseClientServer:
+        if self.test_case_id.info.is_same_node:
+            return self.conf_client
+        else:
+            return self.conf_server
+
+    def conf_clientserver(self, task_role: TaskRole) -> testConfig.ConfBaseClientServer:
```

Comment:
Would there be a better name for this? `conf_clientserver` doesn't really tell me what I can expect from this function?

I would expect NodeName as part of the function name,  unless there is a good reason for it to be omitted.

Reply:
NodeName doesn't seem good, because it's not the name. It's the configuration of the entire node.

Reply:
How about ClientOrServerConfByTaskRole? WDYT?

Reply:
renaming should be done. I'd rather do that in a follow-up.